krabs/krabs/filtering/comparers.hpp

```diff
@@ -45,9 +36,9 @@ namespace krabs { namespace predicates {
             template <typename Iter1, typename Iter2>
             bool operator()(Iter1 begin1, Iter1 end1, Iter2 begin2, Iter2 end2) const
             {
-                // semantics of boost::contains() require that empty range is always contained
-                return std::distance(begin2, end2) == 0 ||
-                    std::search(begin1, end1, begin2, end2, Comparer()) != end1;
+                // empty test range always contained, even when input range empty
+                return std::distance(begin2, end2) == 0 
```

Comment:
Might be marginally more performant to use ``begin2 == end2`` as the first part of this test instead of ``std::distance(begin2, end2) == 0``.

Reply:
Can't argue with you there; no reason for the additional function call.